examples/all_in_one/nginx/conf/nginx.conf

```
@@ -0,0 +1,39 @@
+events {
+  worker_connections  1024;
+}
+
+http {
+    server_tokens off;
+    include       mime.types;
+    charset       utf-8;
+
+    log_format upstream_time_with_content_class_slo_info  '$remote_addr - $remote_user [$time_local] '
+                                                'request-method="$request_method" request-uri="$request_uri" request_query="$args"'
+                                                ' $status $body_bytes_sent '
+                                                '"$http_referer" uag="$http_user_agent" '
+                                                '"$http_x_forwarded_for" ua="$upstream_addr" '
+                                                'rt="$request_time"';
+
+    access_log /nginx/logs/access_log  upstream_time_with_content_class_slo_info;
+
+    limit_req_zone $binary_remote_addr zone=one:10m rate=1r/m;
+
+    server {
+        server_name   localhost;
+        listen        0.0.0.0:8080;
+
+        set $content_class static;
+        location / {
+          return 200;
+        }
+
+        location /err {
+          return 500;
+        }
+
+        location /drop {
+          # delay incoming requests so that the client will timeout
+          limit_req zone=one burst=5;
```

```suggestion
          limit_req zone=one;
```

why `burst=5`? It is not necessary for this example.

---

The whole point why Nginx rate-limiting is configured is to be able to generate requests which violates latency SLO.

The `burst` directive enables us to do that by queuing the excessive requests (otherwise Nginx would immediately terminate them with HTTP 503).

[Docs](https://docs.nginx.com/nginx/admin-guide/security-controls/controlling-access-proxied-http/)

---

you are right